### ISSUE TITLE
Implement snapshot conditions

### ISSUE THREAD

**Issue by jkowalski** [kopia/kopia#1519]
It may be desireable to only take snapshots if certain environmental conditions are met, for example:

* the laptop is on particular network (DNS, Wifi SSID, IP range, etc.)
* we are not on battery power
* free space on disk X is >Y
* disk X is mounted
* CPU usage is below X
* free RAM is above X
* a particular process is running / not running
* user-provided script returns success (similar to [Actions](https://kopia.io/docs/advanced/actions/))

The checks could be elegantly handled using the same mechanism as policies, but figuring extensible UI and CLI will be tricky. Also tricky - how do we support most of these checks using minimal dependencies and in a secure way?

**Comment by benlk** [comment]
Would aborting a backup be possible with the existing "Before Snapshot" script feature?

**Comment by thom4parisot** [comment]
Yeah, it would be interesting to avoid backuping while being connected on a cellphone hotspot with limited data. Definitely a must have!
